test/runtests.jl

```diff
@@ -1,45 +1,55 @@
 using Convex
-using Convex: DotMultiplyAtom
+using Convex.ProblemDepot: run_tests
 using Test
-using ECOS
-using SCS
-using GLPKMathProgInterface
-using Random
-
-import LinearAlgebra.eigen
-import LinearAlgebra.I
-import LinearAlgebra.opnorm
-import Random.shuffle
-import Statistics.mean
+using SCS, ECOS, GLPK
 
-TOL = 1e-3
-eye(n) = Matrix(1.0I, n, n)
+using MathOptInterface
+const MOI = MathOptInterface
+const MOIU = MOI.Utilities
 
 # Seed random number stream to improve test reliability
+using Random
 Random.seed!(2)
 
-solvers = Any[]
-
-push!(solvers, ECOSSolver(verbose=0))
-push!(solvers, GLPKSolverMIP())
-push!(solvers, SCSSolver(verbose=0, eps=1e-6))
-
-# If Gurobi is installed, uncomment to test with it:
-#using Gurobi
-#push!(solvers, GurobiSolver(OutputFlag=0))
+@testset "ProblemDepot" begin
+    @testset "Problems can run without `solve!`ing if `test==false`; T=$T" for T in (Float64, BigFloat)
+        Convex.ProblemDepot.foreach_problem() do name, func
+            @testset "$name" begin
+                # We want to check to make sure this does not throw
+                func(Val(false), 0.0, 0.0, T) do problem
+                    @test problem isa Convex.Problem{T} # check numeric type
+                    model = MOIU.MockOptimizer(MOIU.Model{T}())
+                    Convex.load_MOI_model!(model, problem) # make sure it loads without throwing
+                end
+            end
+        end
+    end
+end
 
-# If Mosek is installed, uncomment to test with it:
-#using Mosek
-#push!(solvers, MosekSolver(LOG=0))
+@testset "Convex with MOI" begin
+    include("test_moi.jl")
+end
 
 @testset "Convex" begin
     include("test_utilities.jl")
-    include("test_const.jl")
-    include("test_affine.jl")
-    include("test_lp.jl")
-    include("test_socp.jl")
-    include("test_sdp.jl")
-    include("test_exp.jl")
-    include("test_sdp_and_exp.jl")
-    include("test_mip.jl")
+
+    @testset "SCS" begin
+        run_tests(; exclude=[   r"mip",
```

Comment:
Thanks very much for the review! I kind of prefer the current way, since it makes it clear that the bug is only with `sdp_matrix_frac_atom`; as you might know, the fact that SCS cannot handle mixed-integer problems is just a limitation of SCS (which we used to gate around by `can_solve_mip` conditionals in the tests-- now with the new system, we have to be explicit, which I think is an advantage). What is the advantage of putting the comment above?

Reply:
> What is the advantage of putting the comment above?

Just that it makes the actual code easier to read and doesn't require lining up the elements of the array in a certain way.

Reply:
Ok, fair enough. I moved the comment up but added a few words to say that it's the `sdp_matrix_frac_atom` which is the test which hits the SCS bug. Hopefully then we have the best of both.